presto-native-execution/src/test/java/com/facebook/presto/nativeworker/PrestoNativeQueryRunnerUtils.java

```diff
@@ -124,36 +115,52 @@ public static class HiveQueryRunnerBuilder
+        // properties it should be created once all the other query runner configs are set. This variable indicates
+        // whether the query runner returned by builder should use an external worker launcher, it will be true only
+        // for the native query runner and should NOT be explicitly configured by users.
         private boolean useExternalWorkerLauncher;
 
-        private HiveQueryRunnerBuilder(String security,
-                                       Map<String, String> extraProperties,
-                                       Map<String, String> extraCoordinatorProperties,
-                                       Map<String, String> hiveProperties,
-                                       Map<String, String> tpcdsProperties,
-                                       boolean useExternalWorkerLauncher)
+        private HiveQueryRunnerBuilder(QueryRunnerType queryRunnerType)
         {
-            this.security = security;
-            this.useExternalWorkerLauncher = useExternalWorkerLauncher;
-            this.extraProperties = extraProperties;
-            this.extraCoordinatorProperties = extraCoordinatorProperties;
-            hiveProperties.put("hive.storage-format", storageFormat);
-            this.hiveProperties = hiveProperties;
-            this.tpcdsProperties = tpcdsProperties;
+            if (queryRunnerType.equals(QueryRunnerType.NATIVE)) {
+                this.extraProperties.putAll(getNativeWorkerSystemProperties());
+                this.extraCoordinatorProperties.putAll(ImmutableMap.<String, String>builder()
+                        .put("native-execution-enabled", "true")
+                        .put("http-server.http.port", "8081")
+                        .build());
+                // The property "hive.allow-drop-table" needs to be set to true because security is always "legacy" in NativeQueryRunner.
+                this.hiveProperties.putAll(ImmutableMap.<String, String>builder()
+                        .putAll(getNativeWorkerHiveProperties())
+                        .put("hive.storage-format", this.storageFormat)
+                        .put("hive.allow-drop-table", "true")
+                        .build());
+                this.tpcdsProperties.putAll(getNativeWorkerTpcdsProperties());
+                this.security = "legacy";
+                this.useExternalWorkerLauncher = true;
+            }
+            else {
+                this.extraProperties.putAll(ImmutableMap.of(
+                        "regex-library", "RE2J",
+                        "offset-clause-enabled", "true"));
+                this.hiveProperties.putAll(ImmutableMap.<String, String>builder()
+                        .put("hive.storage-format", this.storageFormat)
+                        .put("hive.pushdown-filter-enabled", "true")
+                        .build());
+                this.tpcdsProperties.putAll(getNativeWorkerTpcdsProperties());
```

Comment:
Its a bit odd this is getNativeWorkerTpcdsProperties as well. Please can you confirm.

You could write this code to move the common properties for the query runner outside of the if condition.

Reply:
`getNativeWorkerTpcdsProperties()` has config `tpcds.use-varchar-type=true`, which is needed in the java query runner to create Tpcds tables with Varchar columns for the native Tpcds e2e tests.
Moved the common properties outside the if condition, thanks for the suggestion! Could you please take another look?